src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1070,29 +1052,6 @@ void CodeGen::genSSE41Intrinsic(GenTreeHWIntrinsic* node)
 
     switch (intrinsicID)
     {
-        case NI_SSE41_CeilingScalar:
-        case NI_SSE41_FloorScalar:
-        case NI_SSE41_RoundCurrentDirectionScalar:
-        case NI_SSE41_RoundToNearestIntegerScalar:
-        case NI_SSE41_RoundToNegativeInfinityScalar:
-        case NI_SSE41_RoundToPositiveInfinityScalar:
-        case NI_SSE41_RoundToZeroScalar:
-        {
-            assert((baseType == TYP_FLOAT) || (baseType == TYP_DOUBLE));
-            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, node->gtSIMDBaseType);
-
-            if (op2 == nullptr)
```

Comment:
Where is the handling for `op2 == nullptr`?

Currently, there is no codepath that will cover `R_R_R_I`

Reply:
Now, `numArgsOfHWIntrinsics` automatically return the correct `numArgs`, which 1-arg from will be addressed in its own path.

Reply:
>  there is no codepath that will cover R_R_R_I (for the 1 argument case)

This is automatically solved by `IsDstSrcSrcAVXInstruction`.